final_task/rss_reader/rss_reader.py

```diff
@@ -0,0 +1,267 @@
+    parser.add_argument('--date', type=str, help='Specifies the date of news')
+    return parser.parse_args()
+
+
+def write_cache(news):
+    flag = True
+    exists = True
+    try:
+        open('cache.json')
+    except FileNotFoundError:
+        exists = False
+    if exists:
+        json_list = json.load(open('cache.json'))
+        for item in news:
+            for elem in json_list:
+                if item['title'] == elem['title']:
+                    flag = False
+                    break
+            if flag:
+                json_list.append(item)
+        with open("cache.json", "w") as file:
+            json.dump(json_list, file, indent=2, ensure_ascii=False)
+    else:
+        with open("cache.json", "w") as file:
+            json.dump(news, file, indent=2, ensure_ascii=False)
+
+
+
+
+def get_by_date(date, source, limit):
+    flag = False
+    calendar = {
+        "01": 'Jan',
+        "02": 'Feb',
+        "03": 'Mar',
+        "04": 'Apr',
+        "05": 'May',
+        "06": 'Jun',
+        "07": 'Jul',
+        "08": 'Aug',
+        "09": 'Sep',
+        "10": 'Oct',
+        "11": 'Nov',
+        "12": 'Dec',
+    }
+    date = date[6:] + ' ' + calendar[date[4:6]] + ' ' + date[0:4]
+    try:
+        file = open('cache.json')
+    except FileNotFoundError:
+        print('No news in cache yet')
+    else:
+        with file:
+            json_list = json.load(file)
+            print(date + ':\n')
+            if source:
+                source = source[8:source.find('/', 8)]
+                for item in json_list:
+                    if date in item['date'] and source in item['link']:
+                        flag = True
+                        print_one(item)
+                        limit -= 1
+                        if not limit:
+                            break
+                else:
+                    for item in json_list:
+                        if date in item['date']:
+                            flag = True
+                            print_one(item)
+                            limit -= 1
+                            if not limit:
+                                break
+            if not flag:
+                log.error('No news on ' + date + ' have been found')
+                print('No news on ' + date + ' have been found')
+
+
+if __name__ == '__main__':
+    try:
+        log.info("Main")
+        args = args_parse()
+        if is_connected():
+            if args.version:
+                log.info(f'Version {args.version}')
+                print('Version is 1.2')
+            if args.verbose:
+                logging.basicConfig(level=logging.INFO)
+
+            if args.limit:
+                limit = args.limit
+                log.info(f'Limit = {limit}')
+            if args.source:
+                if 'www' in args.source:
+                    args.source = args.source[:8] + args.source[12:]
```

Comment:
A question:

Why do you need to do this?

Reply:
Because a user can type both https://www.lenta.ru/rss and https://lenta.ru/rss
And I need a single variant of url to compare it with other links when I use source + date options
So I remove "www." if I find it in url

Reply:
It is better to just shutdown the program if the user specified incorrect link or the link that does not return rss feed.
There may be rss feed links that may contain `www` naturally